sqlmesh/core/context.py

```diff
@@ -1321,7 +1322,12 @@ def table_diff(
             if not on:
                 for ref in model.all_references:
                     if ref.unique:
-                        on = ref.columns
+                        expr = ref.expression
+
+                        if isinstance(expr, exp.Tuple):
+                            on = [key.this.sql() for key in expr.expressions]
+                        else:
+                            on = [expr.this.sql()]
```

Comment:
Let's add a test to cover this code path.

Are we sure the only alternative here is an `exp.Paren`? If that's the case, I'd add an `assert isinstance(expr, exp.Paren)` here to enforce this and make it clear what case this `else` is handling.

Reply:
it's not only `exp.Paren` the grain might also be defined without a parenthesis so in that case an `exp.Column`, but in both cases `.sql()` will transform them into the correct sql, this is why I removed the conditional. 

This code path is tested in: `tests/integrations/jupyter/test_magics.py::test_table_diff`, but I can also add another test explicitly for that

Reply:
> it's not only exp.Paren the grain might also be defined without a parenthesis so in that case an exp.Column, but in both cases .sql() will transform them into the correct sql, this is why I removed the conditional.

Sounds good.

> This code path is tested in: tests/integrations/jupyter/test_magics.py::test_table_diff, but I can also add another test explicitly for that

Gotcha, should be fine then. I see now that we diff `sushi.top_waiters` which has a single column for its grain.